### DISCUSSION TITLE
Entity Export for Aubess Smart Plug+ Wifi

### DISCUSSION THREAD

**Discussion by chicchik** [make-all/tuya-local#2580] [General]
Good morning, this is my first time here.
I have recently purchased WIFI Smart Plugs (purchased cheaply on Aliexpress,identified as “Smart Plug +”, product category “tdbq”, but otherwise there is no reference to the manufacturer) that are integrated on Smart Life App and that I would like to integrate into my Home Assistant.
They are detected in Home Assistant either through Tuya integration or Local Tuya, but in both cases the only entity exposed is the switch entity and then it is limited to on or off. So I tried the Tuya Local integration (which identifies them as aubess_1gang_switch when acquiring them), which actually exposes all the values I'm interested in (power, current, and voltage) but I would also need a consumption totalizer (present instead in the Smart Life app) so that I could then integrate it with the energy dashboard.
In the device settings log, taken from tuya's cloud developer, among the various “DP IDs” there is one called “Add Electricity” that passes to the cloud every 10 minutes a value in Kwh as if to provide a consumption information for those 10 minutes that is then processed by the app to total the consumption (of the day and the month). Do you know if it is possible to leverage this “DP ID” to pass the information to Home assistant and thus allow it to total the consumption?

If there are no solutions to this, does anyone know if the same ZIGBEE sockets integrated through Z2mqtt provide the totalizers I requested?
Thanks in advance to all for your answers and comments.

**Comment by make-all** [comment]
It is better to use an Integration helper to calculate based on the power readings.

As you observed, the "add_ele" dp on tuya smartplugs only reports either energy consumption since the last report on a regular schedule (once every 10 minutes) or it reports 1 every time 0.1 kWh ticks over.  Neither of these is particularly easy to turn into a cumulative energy consumption with the standard helpers in HA.
